File: leetcode/11.container-with-most-water.py
#
# @lc app=leetcode id=11 lang=python3
#
# [11] Container With Most Water
#

# @lc code=start
class Solution:
	def maxArea(self, height: List[int]) -> int:
		front, back, area = 0, len(height) - 1, 0
		while (back > front):
			area = max(area, min(height[back], height[front]) * (back - front))
			if (height[back] < height[front]):
				back -= 1
			else:
				front += 1
		return area
# @lc code=end

File: leetcode/11.container-with-most-water.cpp
/*
 * @lc app=leetcode id=11 lang=cpp
 *
 * [11] Container With Most Water
 */

// @lc code=start
#include <algorithm>
#include <vector>
#include <>

using namespace std;

class Solution {
public:
	int maxArea(vector<int>& height) {
		auto	tallest = max_element(height.begin(), height.end());
		int		max_potential = *tallest * height.size();
		int		max_area = 0;

		for (auto it = height.begin(); it != height.end(); it++) {
			
		}
	}
};
// @lc code=end

